mongodump/oplog_dump.go

```diff
@@ -104,13 +103,14 @@ func oplogDocumentFilter(in []byte) ([]byte, error) {
 		return nil, fmt.Errorf("cannot dump with oplog if admin.system.version is modified")
 	}
 
-	for i := range rawD {
-		if rawD[i].Name == "o" {
-			var rawO bson.RawD
-			err = bson.Unmarshal(rawD[i].Value.Data, &rawO)
-			for j := range rawO {
-				if rawO[j].Name == "renameCollection" {
-					return nil, fmt.Errorf("cannot dump with oplog while renames occur")
+	for _, e := range rawD {
```

Comment:
I know this is copy/paste code, but this might be cleaner if we decode into a bson.M instead of a bson.D so we can do lookups instead of nested for loops.

Reply:
Yeah, this for-looping code is pretty bad. I never took a step back to look at it. I've taken your suggestion one step further and just updated the struct that we're already unmarshalling to with an "o.renameCollection" field.

Reply:
This has since been changed to using `Raw`'s `LookupErr` instead of partial unmarshalling.